### ISSUE TITLE
Apothecary Pouch/Component Pouch deleting your items

### ISSUE THREAD

**Issue by TheDickstaple** [MysticMods/Roots#185]
Sometimes if I have two stacks of pretty much anything in the same pouch, I'll grab part of a stack to do something, then when I'm finished I'll open my pouch and try to deposit what's left of what I previously grabbed back into it's original stack, the amount adds up but when I close the pouch and then re-open it again, whatever I added to that stack somehow disappeared as if the action never happened. I have no clue why this occurs, I tried simply moving half the stack into a different slot, that slot then became empty after closing and opening the pouch again, making me lose half of that resource. It's a pretty annoying inconvenience and drains my resources, I'm using the latest version (3.0.0 1.12.2-3.0.7)

**Comment by duely** [comment]
Thank you for the exact reproduction steps! I was worried that this bug might be lingering but I didn't have any specifics on how to properly test it, and none of my tests seemed to cause it. I'll dig into it!

**Comment by TheDickstaple** [comment]
If it doesn't happen with Roots on it's own then it's most likely an issue with the combination of mods my girlfriend and I use on our server. I'll list them if it becomes necessary.

**Comment by TheDickstaple** [comment]
Little update because I'd been falling asleep as I discovered the bug, from what I can tell it seems that I have a way to go around it, in this case I have a full stack of pereskia bulbs in my pouch and I have three more bulbs as well, if I want to put in different new items without them disappearing the moment I close the pouch, I had to grab the incomplete stack of pereskia bulbs and try to combine it with the already full one, which obviously does nothing but then when I put the incomplete stack back down, no items are lost.

**Comment by duely** [comment]
Which sounds very much to me as though the sync-to-stack functions aren't triggering again for some reason. Damn it! I'm looking into this currently.

**Comment by duely** [comment]
I'm unable to replicate it in single-player, where I did most of my testing, leading me to think that perhaps this is a server/client issue that's being (badly) bypassed in single-player mode -- the fact that you mentioned it specifically on a server leads me to think this is the case (as most of the other times I experienced this was on a server, but never replicated in single-player). Digging into it more!

**Comment by duely** [comment]
Okay, able to replicate it on a test server! I'll see what I can do about it.

**Comment by duely** [comment]
I think I've managed to fix it. I'll be doing a new release within the next day or so, I'd appreciate it if you'd be able to check it out and test to see whether the functionality has changed! I noticed that it was very specifically when trying to merge stacks by clicking on one item, then clicking it into the stack -- whereas I tend to shift-click almost everything. That's possibly why I didn't notice it until your report.
